Tests/Packet++Test/Tests/HttpTests.cpp

```diff
@@ -339,7 +348,7 @@ PTF_TEST_CASE(HttpResponseLayerParsingTest)
 	pcpp::HttpResponseLayer* responseLayer = httpPacket.getLayerOfType<pcpp::HttpResponseLayer>();
 	PTF_ASSERT_NOT_NULL(responseLayer);
 
-	PTF_ASSERT_EQUAL(responseLayer->getFirstLine()->getStatusCode(), pcpp::HttpResponseLayer::Http200OK, enum);
+	PTF_ASSERT_EQUAL(responseLayer->getFirstLine()->getStatusCode(), pcpp::HttpResponseStatusCode::Http200OK, enum);
```

Comment:
We don't have to change this line, it will work even if we don't change it.

Same as for the other changes below this line. I think we should preserve it as is to ensure backward compatibility

Reply:
previous Http200OK is a global enum in `HttpResponseLayer` namespace; however, unfortunately, now the enum type is actually `HttpResponseStatusCode::Value` (or `HttpResponseLayer::HttpResponseStatusCode::Value`, we cannot use the enum without `HttpResponseStatusCode::`.

it's okay to do the following:
- `pcpp::HttpResponseLayer::Http200OK`
- `pcpp::HttpResponseLayer::HttpResponseStatusCode::Http200OK`
- `pcpp::HttpResponseLayer::HttpResponseStatusCode::Value::Http200OK`

IMO, it's also clearer to use `pcpp::HttpResponseStatusCode::Http200OK` rather than `pcpp::HttpResponseLayer::Http200OK`. what do you think @seladb ?

Reply:
however, I can think of one way, to expose `HttpResponseStatusCode::Value` under `HttpResponseLayer` ...
I personally really don't like this ...

Reply:
I generally agree with you, my only problem is that it breaks the backward compatibility that we tried to solve with this:
https://github.com/seladb/PcapPlusPlus/pull/1180/files#diff-30f6094c6b01047985499a354425a4f628eadc54a0b3b822c513ec95f72d26a6R485

Meaning, that if someone is already using `pcpp::HttpResponseLayer::Http200OK` their code breaks once they update to the latest code

Reply:
@tigercosmos is there anything we can do with that? ⬆️

Reply:
@seladb 

we can still achieve backward compatibility by using the following code in `HttpResponseLayer` :
```
using Http100Continue = pcpp::HttpResponseStatusCode::Http100Continue;
using Http101SwitchingProtocols = pcpp::HttpResponseStatusCode::Http101SwitchingProtocols;|
```

It's possible, of course; however, the code will look very redundant and trivial.

the current latest version is v22.11, I think we can check in this patch in v23 (or apply the above code first and remove in the v23), so that people can have a common sense that users may need to migrate(update) the code themself due to the breaking change.

Reply:
Yes, I guess you're right. Even though it breaks backward compatibility the fix is pretty simple, basically replace `pcpp::HttpResponseLayer::Http200OK` with `pcpp::HttpResponseStatusCode:Http200OK` or `pcpp::HttpResponseLayer::HttpResponseStatusCode:Http200OK`.

Let's keep it like this